Tests/EventsTest.cs

```diff
@@ -217,6 +217,40 @@ public void TestIdentifyTNoUserId()
             Assert.Equal(expectedUserId, actualUserId);
         }
 
+
+        [Fact]
+        public void TestIdentifyReload()
+        {
+            string expectedUserId = "newUserId";
+            var actualIdentify = new List<IdentifyEvent>();
+            var actualTrack = new List<TrackEvent>();
+            _plugin.Setup(o => o.Identify(Capture.In(actualIdentify)));
+            _plugin.Setup(o => o.Track(Capture.In(actualTrack)));
+
+            _analytics.Add(_plugin.Object);
+            _analytics.Identify(expectedUserId);
+
+            string actualUserId = _analytics.UserId();
+
+            Assert.NotEmpty(actualIdentify);
+            Assert.Equal(expectedUserId, actualIdentify[0].UserId);
+            Assert.Equal(expectedUserId, actualUserId);
+
+            var userIdFilled = UserInfo.DefaultState(_analytics.Storage);
+            Assert.Equal(expectedUserId, userIdFilled._userId);
+
+            // Now validate null ID get set            
+            _analytics.Identify(null, null);
+
+            string emptyUserId = _analytics.UserId();
```

Comment:
we should also verify the storage actually changed. querying from `analytics` only checks the in memory copy.